feat: Add configurable `artifactErrorWarning` user string

hesstobi opened the pull request:
## Changes

This PR introduces a new configurable user string `artifactErrorWarning` that allows users to customize the warning message displayed in PR comments when artifact errors occur during dependency updates.

**Key changes:**
- Added `artifactErrorWarning` to the `userStrings` configuration object with default value: `"You probably do not want to merge this PR as-is."`
- Modified artifact error comment generation in `lib/workers/repository/update/branch/index.ts` to use the configurable user string with template compilation support
- Updated documentation to include the new user string option with examples
- Added comprehensive test coverage for default, custom, and templated messages

## Context

This change was implemented to support GitLab's quick action functionality, specifically to enable adding labels using the `/label ~my-nice-label` command in PR comments. By making the artifact error warning message configurable, users can now include GitLab quick actions directly in their custom warning messages, allowing for automated label management when artifact errors occur.

**Example use case:**
```json
{
  "userStrings": {
    "artifactErrorWarning": "You probably do not want to merge this PR as-is.\n\n/label ~artifact-error ~needs-review"
  }
}
```

This enables automatic labeling of PRs with artifact errors for improved workflow management in GitLab environments.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

**Test coverage includes:**
- Verification of default message behavior (backward compatibility)
- Testing custom user-defined messages
- Testing template variable substitution (e.g., `{{manager}}`)

tobias-westerheide committed:
feat(config): add artifactErrorWarning user string option

Add new userStrings.artifactErrorWarning configuration option to allow
users to customize the warning message displayed when artifact errors
occur during updates.

Default message: 'You probably do not want to merge this PR as-is.'"

tobias-westerheide committed:
"feat(branch): use configurable artifact error warning message

Replace hardcoded artifact error warning text with user-configurable
message from userStrings.artifactErrorWarning. The message supports
template compilation for dynamic content.

This allows users to customize the warning shown in PR comments when
lock file or other artifact generation fails."

tobias-westerheide committed:
test(branch): add tests for configurable artifact error warnings

Add test cases covering:
- Default artifact error warning message behavior
- Custom user-configured warning messages
- Template variable substitution in warning messages

Ensures proper functionality of the new artifactErrorWarning
configuration option."

tobias-westerheide committed:
docs: document artifactErrorWarning user string option

Add documentation for the new userStrings.artifactErrorWarning
configuration option, including description and usage example.

This option allows users to customize the warning message displayed
when artifact errors occur during dependency updates.

hesstobi commented:
Hi team!

I've implemented this feature to enable GitLab quick actions (like `/label ~my-nice-label`) in artifact error comments, which should improve workflow automation for GitLab users.

I'd appreciate any feedback on the implementation:
- Is the approach with `userStrings` the right way to go for this type of customization?
- Are there any edge cases I should consider or additional test scenarios needed?
- Should I add any additional documentation or examples?
- Any other improvements or changes you'd like to see?

I'm happy to make any adjustments or additions based on your review!
Thanks for taking the time to review this PR.

tobias-westerheide committed:
Merge branch 'main' into custom-messages-on-articate-update-comments

tobias-westerheide committed:
Merge branch 'main' into custom-messages-on-articate-update-comments

RahulGautamSingh commented on lib/workers/repository/update/branch/index.spec.ts:
```diff
@@ -1103,6 +1103,88 @@ describe('workers/repository/update/branch/index', () => {
       commit.commitFilesToBranch.mockResolvedValueOnce(null);
       await branchWorker.processBranch(config);
       expect(platform.ensureComment).toHaveBeenCalledTimes(1);
```
```suggestion
```
The new assertion covers this

RahulGautamSingh commented on lib/workers/repository/update/branch/index.spec.ts:
```diff
@@ -1103,6 +1103,88 @@ describe('workers/repository/update/branch/index', () => {
+        },
+      };
+      await branchWorker.processBranch(inconfig);
+      expect(platform.ensureComment).toHaveBeenCalledTimes(1);
```
```suggestion
```

RahulGautamSingh commented on lib/workers/repository/update/branch/index.spec.ts:
```diff
@@ -1103,6 +1103,88 @@ describe('workers/repository/update/branch/index', () => {
+        },
+      };
+      await branchWorker.processBranch(inconfig);
+      expect(platform.ensureComment).toHaveBeenCalledTimes(1);
```
```suggestion
```

RahulGautamSingh commented on lib/workers/repository/update/branch/index.ts:
```diff
@@ -870,11 +870,13 @@ export async function processBranch(
           { artifactErrors: config.artifactErrors },
           'artifactErrors',
         );
+        // TODO #22198
+        const userStrings = config.userStrings!;
```
```suggestion
```

RahulGautamSingh commented on lib/workers/repository/update/branch/index.ts:
```diff
@@ -870,11 +870,13 @@ export async function processBranch(
-        content +=
-          ' related to this branch. You probably do not want to merge this PR as-is.';
+        content += ' related to this branch. ';
+        content += template.compile(userStrings.artifactErrorWarning, config);
```
```suggestion
        content += template.compile(config.userStrings!.artifactErrorWarning, config);
```

RahulGautamSingh reviewed (changes requested):
Minor nitpicking. All else looks good.

tobias-westerheide committed:
refactor: simplify test assertions and inline variable assignment

- Remove redundant toHaveBeenCalledTimes assertions in branch tests
- Inline userStrings variable assignment in artifact error handling

hesstobi commented:
@RahulGautamSingh Thank you for your review. I added your suggestion.
